src/main/java/org/jboss/pnc/konfluxtooling/logging/UploadLogCommand.java

```diff
@@ -0,0 +1,89 @@
+package org.jboss.pnc.konfluxtooling.logging;
+
+import java.io.File;
+import java.math.BigInteger;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.MessageDigest;
+import java.time.OffsetDateTime;
+
+import org.jboss.pnc.bifrost.upload.BifrostLogUploader;
+import org.jboss.pnc.bifrost.upload.LogMetadata;
+import org.jboss.pnc.bifrost.upload.TagOption;
+
+import io.quarkus.logging.Log;
+import picocli.CommandLine;
+
+@CommandLine.Command(name = "upload-log")
+public class UploadLogCommand implements Runnable {
+
+    private static final int DEFAULT_MAX_RETRIES = 4;
+
+    private static final int DEFAULT_DELAY_SECONDS = 60;
+
+    @CommandLine.Option(names = "--file", required = true)
+    String logFile;
+
+    @CommandLine.Option(names = "--bifrost-url", required = true)
+    String bifrostURL;
+
+    @CommandLine.Option(names = "--max-retries")
+    int maxRetries = DEFAULT_MAX_RETRIES;
+
+    @CommandLine.Option(names = "--delay-seconds")
+    int delaySeconds = DEFAULT_DELAY_SECONDS;
+
+    @CommandLine.Option(names = "--process-context", required = true)
+    String processContext;
+
+    @CommandLine.Option(names = "--process-context-variant", required = true)
+    String processContextVariant;
+
+    @CommandLine.Option(names = "--tmp", required = true)
+    String tmp;
+
+    @CommandLine.Option(names = "--request-context", required = true)
+    String requestContext;
+
+    public void run() {
+        try {
+            var logFilePath = Path.of(logFile);
+            var file = logFilePath.toFile();
+            if (!file.exists()) {
+                throw new RuntimeException(String.format(
+                        "No log file found at %s. Has the build been correctly done?", logFilePath));
```

Comment:
Following on from https://github.com/project-ncl/konflux-tooling/pull/11/files#r1896576530 while currently the `deployment` directory is passed in https://github.com/redhat-appstudio/jvm-build-service/blob/main/deploy/tasks/maven-deployment.yaml#L104 but perhaps it should be the lower level workdir directory which should contain deployment and log directories next to each other?

Reply:
do you mean something like this? 
/var/workdir/results
  |-- deployment
  |-- log
@matejonnet wdty?

Reply:
No just /var/workdir but this only makes sense if everything is in the same tool